src/ui/page/share/dataset/steps/metadataFilling/metadataFilling.tsx

```diff
@@ -66,47 +99,58 @@ export const MetadataFilling: FC = () => {
 
   const handleFieldValueChange = useCallback(
     (id: string) => (value: string) => {
-      setFormItemValue({ id, value })()
+      setFormItemValue(id, value)()
     },
     [setFormItemValue]
   )
 
-  const tagsFieldValue = useCallback(
-    (id: string): string[] =>
-      flow(
-        formItemById,
-        IOO.map(({ value }: FormItem) => value),
-        IOO.flatMap(IOO.fromPredicate(Array.isArray)),
-        IOO.getOrElseW(() => () => []),
-        apply(null)
-      )(id),
-    [formItemById]
-  )
-
-  const addTag = useCallback(
+  const handleNumericValueChange = useCallback(
     (id: string) => (value: string) => {
-      setFormItemValue({ id, value: A.append(value)(tagsFieldValue(id)) })()
+      !isNaN(Number(value)) && setFormItemValue(id, Number(value))()
     },
-    [setFormItemValue, tagsFieldValue]
+    [setFormItemValue]
   )
 
-  const removeTag = useCallback(
-    (id: string) => (value: string) => {
-      const filterPredicate = (tag: string): boolean => tag !== value
-      setFormItemValue({ id, value: A.filter(filterPredicate)(tagsFieldValue(id)) })()
-    },
-    [setFormItemValue, tagsFieldValue]
-  )
+  const numericalFieldValue = (v: DatasetFormItem['value']) => (): string =>
+    pipe(
+      v,
+      O.flatMap(O.fromPredicate(N.isNumber)),
```

Comment:
it's not working without because `v` is an `Option`. is it another way to handle it?

Reply:
I see. So, the `fromPredicate` is redundant. Because given an `Option` of something (e.g. `v`), you want to get an `Option` of `boolean`. This is where the `map`can help to convert the value with the predicate function (e.g. `N.isNumber`):

```ts
pipe(
      v,
      O.map(N.isNumber),
      O.fold(
        () => '',
        n => n.toString()
      )
    )
```

Reply:
@lolottetheclash it's ok. you can leave it like that. :relaxed: